src/Time/Actions/ChangeStepSize.hpp

```diff
@@ -53,63 +127,31 @@ struct ChangeStepSize {
   template <typename DbTags, typename... InboxTags, typename Metavariables,
             typename ArrayIndex, typename ActionList,
             typename ParallelComponent>
-  static std::tuple<db::DataBox<DbTags>&&> apply(
+  static std::tuple<db::DataBox<DbTags>&&, bool, size_t> apply(
       db::DataBox<DbTags>& box, tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
       const Parallel::GlobalCache<Metavariables>& cache,
       const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
       const ParallelComponent* const /*meta*/) noexcept {
     static_assert(Metavariables::local_time_stepping,
                   "ChangeStepSize can only be used with local time-stepping.");
-
-    const LtsTimeStepper& time_stepper = db::get<Tags::TimeStepper<>>(box);
-    const auto& step_choosers = Parallel::get<step_choosers_tag>(cache);
-    const auto& step_controller = Parallel::get<Tags::StepController>(cache);
-
-    const auto& time_id = db::get<Tags::TimeStepId>(box);
-    const auto& history = db::get<Tags::HistoryEvolvedVariables<>>(box);
-
-    if (not time_stepper.can_change_step_size(time_id, history)) {
-      return std::forward_as_tuple(std::move(box));
+    const bool step_successful =
+        change_step_size<StepChooserRegistrars>(make_not_null(&box), cache);
+    if (step_successful) {
+      return {std::move(box), false,
+              tmpl::index_of<ActionList, ChangeStepSize>::value};
+    } else {
+      if constexpr (tmpl::none<ActionList, tt::is_a_lambda<Actions::UpdateU,
+                                                           tmpl::_1>>::value) {
+        ERROR(
+            "Step not successful, and there is no UpdateU action to return "
+            "to.");
```

Comment:
One last question for my understanding: What is this case supposed to handle?  If the run is going to die because the chosen step size is a bit off, should it at least give some information about what is wrong?  Do we really want to abort in this case instead of continuing with a warning or something?

Reply:
This is intended to flag a malformed action list - effectively, what I'm trying to communicate is that if the `ChangeStepSize` action is used, the `UpdateU` action must also be used to permit the action-list control flow (instead of stepping as a subroutine e.g. in `ComputeTimeDerivatives`)
But, now that you point it out, I see that this would be better handled with a static assert, so I'll squash that in too.